src/analyses/ai.cpp

```diff
@@ -499,10 +499,11 @@ void ai_baset::sequential_fixedpoint(
   const goto_functionst &goto_functions,
   const namespacet &ns)
 {
-  // do each function at least once
+  goto_functionst::function_mapt::const_iterator
+    f_it=goto_functions.function_map.find(goto_functions.entry_point());
 
-  forall_goto_functions(it, goto_functions)
-    fixedpoint(it->second.body, goto_functions, ns);
+  if(f_it!=goto_functions.function_map.end())
+    fixedpoint(f_it->second.body, goto_functions, ns);
```

Comment:
Apologies for the major delay: Yes, please do move this into a separate commit as it could be contentious for cases where the entry function isn't available (it's not clear to me whether running abstract interpretation considering each function a possible entry point is actually a bad idea).